### PR TITLE
Add info about cider-repl-prompt-function

### LINKED ISSUE #1238: Document cider-repl-prompt-function
`cider-repl-prompt-function` is super useful, but basically no one knows about it. Even I recently forgot about it. It needs an entry in the README.

### PR DESCRIPTION
**Pull request by kleewho**
Closes #1238

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -292,6 +292,12 @@ When using `switch-to-buffer`, pressing <kbd>SPC</kbd> after the command will
 make the hidden buffers visible. They'll always be visible in
 `list-buffers` (<kbd>C-x C-b</kbd>).
 
+* You can customize the prompt in REPL buffer. To do that you can customize
+  `cider-repl-prompt-function` and pass to it function that takes one argument,
```
Comment (Malabarba) [inline README.md]: I think `pass to it` -> `set it to a`.

Comment (kleewho) [inline README.md]: Changed. Thanks

File: README.md [hunk 2]
```diff
@@ -292,6 +292,12 @@ When using `switch-to-buffer`, pressing <kbd>SPC</kbd> after the command will
 make the hidden buffers visible. They'll always be visible in
 `list-buffers` (<kbd>C-x C-b</kbd>).
 
+* You can customize the prompt in REPL buffer. To do that you can customize
+  `cider-repl-prompt-function` and set it to a function that takes one argument,
+  a namespace name. For convenience, three functions are already provided:
+  `cider-repl-prompt-lastname`, `cider-repl-prompt-abbreviated`,
+  `cider-repl-prompt-default` and by default the last one is being used.
```
Comment (bbatsov) [inline README.md]: Maybe we should add here some examples showcasing the built-in functions? And perhaps an example showing custom ns processing?

Comment (kleewho) [inline README.md]: Sure thing. I'll prepare some examples

**Comment by bbatsov** [comment]
P.S. We generally refer issues in the commit title - e.g. `[Fix #1238] ....`.

**Comment by kleewho** [comment]
@bbatsov done. Commit message + examples

**Comment by bbatsov** [comment]
:+1: Thanks!

**Pull request merged**